core/jvm/src/main/scala/zio/IOPlatformSpecific.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import _root_.java.util.concurrent.CompletableFuture
+
+private[zio] trait IOPlatformSpecific {
+
+  def toCompletableFuture[E, A](io: IO[E, A])(implicit ev: E <:< Throwable): UIO[CompletableFuture[A]] =
+    toCompletableFutureWith(io)(ev)
+
+  def toCompletableFutureWith[E, A](io: IO[E, A])(f: E => Throwable): UIO[CompletableFuture[A]] =
```

Comment:
Ditto for this and all other examples. If a method takes an effect as its first parameter, it's a good sign it's better added to the ZIO data type as a method, rather than the companion object.

Reply:
Yes, gotcha. So the only I know of doing mixins on traits this way is by using self-type. Is that what you had in mind or is there another technique which I'm not aware of? i.e.

1. in the JVM branch:

```scala
trait ZIOPlatformSpecific[-R, +E, +A] { self: ZIO[R, E, A] =>
  def toCompletableFutureWith(f: E => Throwable): URIO[R, CompletableFuture[A]] = ???
}
```

2. in the shared branch:

```scala
sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E, A]
```

Reply:
Self-type should be fine!